app/Http/Livewire/Delegates/FavoriteDelegateHandler.php

```diff
@@ -0,0 +1,51 @@
+<?php
+
+declare(strict_types=1);
+
+namespace App\Http\Livewire\Delegates;
+
+use App\Facades\Settings;
+use App\Http\Livewire\Concerns\HandlesSettings;
+use Livewire\Component;
+
+final class FavoriteDelegateHandler extends Component
+{
+    use HandlesSettings;
+
+    /** @var mixed */
+    protected $listeners = [
+        'setFavoriteDelegate'    => 'setDelegate',
+        'removeFavoriteDelegate' => 'removeDelegate',
+    ];
+
+    public function render(): string
+    {
+        return '<div></div>';
+    }
+
+    public function setDelegate(string $publicKey): void
+    {
+        $delegates = Settings::favoriteDelegates();
+
+        if ($delegates->has($publicKey)) {
+            return;
+        }
+
+        $delegates->add($publicKey);
+
+        $this->saveSetting('favoriteDelegates', $delegates->toArray());
+    }
+
+    public function removeDelegate(string $publicKey): void
+    {
+        $delegates = Settings::favoriteDelegates();
+
+        if (! $delegates->has($publicKey)) {
```

Comment:
do we need this check or can we rely on `filter` to handle this scenario already?

Reply:
I added this more to prevent unnecessary updates to the cookies if someone managed to get to the point of removing multiple times